etc/update-latest-release.py

```diff
@@ -1,49 +1,111 @@
 #!/usr/bin/env python3
 
+# 1. Imports
 import os
 import sys
-
-from datetime import datetime
+from datetime import timezone
 from github import Github, Auth
+import yaml
 
-API_KEY = (os.getenv("API_KEY") or os.getenv("GITHUB_TOKEN") or "").strip()
-if API_KEY == None:
-    print("API key was not found! Authentication will fail!\nSet the environment variable API_KEY "
-          "to a github access token and try again!\n")
-    exit()
 
-auth = Auth.Token(API_KEY)
+# 2. Constants
+TARGET_REPO = os.getenv("TARGET_REPO", "oscar-system/Oscar.jl")
+JULIA_MIN = os.getenv("JULIA_MIN", "1.10")
+ownpath = os.path.abspath(sys.argv[0])
+repopath = os.path.dirname(os.path.dirname(ownpath))
+datapath = os.path.join(repopath, "_data")
+releasefilepath = os.path.join(datapath, "release.yml")
+
 
+# 3. Function to write simple key=value outputs for GitHub Actions.
+def gh_output(**kvs):
+    path = os.getenv("GITHUB_OUTPUT")
+    if not path:
+        return
+    with open(path, "a", encoding="utf-8") as f:
+        for k, v in kvs.items():
+            v = "" if v is None else str(v)
+            f.write(f"{k}={v}\n")
+
+# 4. Get/set API key or raise error
+API_KEY = (os.getenv("API_KEY") or os.getenv("GITHUB_TOKEN") or "").strip()
+if not API_KEY:
+    print("API key was not found! Set API_KEY or GITHUB_TOKEN to a GitHub token.")
+    gh_output(changed="false", reason="no-token")
+    sys.exit(1)
+auth = Auth.Token(API_KEY)
 g = Github(auth=auth)
 
-ownpath = os.path.abspath(sys.argv[0])
-repopath = os.path.dirname(os.path.dirname(ownpath))
-datapath = repopath +'/_data'
 
+# 5. Fetch release
 try:
-    repo = g.get_repo("oscar-system/Oscar.jl")
+    repo = g.get_repo(TARGET_REPO)
     release = repo.get_latest_release()
 except Exception as e:
     print(e)
-    print("Network access failed!")
-    print("Leaving the release file unchanged!")
-    exit()
+    print("Network access failed! Leaving the release file unchanged!")
+    gh_output(changed="false", reason="fetch-failed")
+    sys.exit(1)
 
-dt = release.created_at
-version = release.name[1:]
 
-releasestring = f"""\
-version: "{version}"
+# 6. Process release version
+tag = release.tag_name
+if not tag: # Safety fallbacks (rarely needed but harmless)
+    tag = release.name or release.title
+version = (tag or "").lstrip("v").strip()
```

Comment:
```suggestion
version = (tag or "").strip().lstrip("v")
```
As long as we are being safe, we might as well be really safe. `strip()` strips whitepsace from both ends, and then `lstrip()` can strip the v at the beginning.

(Except if you are expecting a string of the form `v<whitespace>x.y.z`, in which case, disregard my comment.)